pkg/controller/cluster/main-controller.go

```diff
@@ -440,6 +449,83 @@ func (c *Controller) GetenvHandler(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// Attempts to fetch given image and then extracts and keeps relevant files (minio, minio.sha256sum & minio.minisig)
+// at a pre-defined location (/tmp/update)
+func (c *Controller) fetchImage(image, tag string) error {
+	ctx := context.Background()
+	srcBasePath := "/usr/bin/"
+	srcBinary := "minio"
+	srcShaSum := "minio.sha256sum"
+	srcSig := "minio.minisig"
+
+	destBasePath := updatePath
+	destBinary := "minio." + tag
+	destShaSum := "minio." + tag + ".sha256sum"
```

Comment:
extract the tag from the `./minio -v` output @nitisht - allows for updating any image

Reply:
The minio binary is inside a container at this point, so it is tricky to run the binary and extract the output here.

Reply:
why not? @nitisht you need to copy them out and use the `os.exec()`  `--version` output as the way to rename them locally

Reply:
done